Add support for DateTime2 DataType in TimeStamp column.

stedel opened the pull request:
This PR adds DateTime2 as a supported underlying DataType for the TimeStamp column.

Fix #300.

ckadluba commented on README.md:
````diff
@@ -436,14 +436,19 @@ This column stores the event level (Error, Information, etc.). For backwards-com
+columnOptions.TimeStamp.DataType = SqlDbType.DateTime2;
+```
+
+Please be aware that you have to configure the sink for `datetimeoffset` if the used logging database table has a `TimeStamp` column of type `datetimeoffset`. If the underlying database uses `datetime2` for the `TimeStamp` column, the sink must be configured to use `datetime2`. On the other hand you must not configure for `datetimeoffset` if the `TimeStamp` column is of type `datetime`. Failing t ...
````
Please replace 

On the other hand you must not configure for `datetimeoffset` if the `TimeStamp` column is of type `datetime`.

with the following

On the other hand you must not configure for `datetimeoffset` if the `TimeStamp` column is of type `datetime` or `datetime2`.

stedel replied:
Integrated your suggestion in d70790f2d6779471bedfe88fdc467985ffe37691.

ckadluba commented on src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/Output/JsonLogEventFormatter.cs:
```diff
@@ -124,7 +124,7 @@ private void WriteTimeStampIfPresent(LogEvent logEvent, TextWriter output, ref s
             precedingDelimiter = _commaDelimiter;
             var colData = WritePropertyName(logEvent, output, StandardColumn.TimeStamp);
-            var value = _columnOptions.TimeStamp.DataType == SqlDbType.DateTime
+            var value = (_columnOptions.TimeStamp.DataType == SqlDbType.DateTime || _columnOptions.TimeStamp.DataType == SqlDbType.DateTime2)
```
Please add a unit test in JsonLogEventFormatterTests.cs.

stedel replied:
Should be resolved in 94ee530b765ac64c166f3001d0e52eb71526f218

stedel commented:
Thanks for you Feedback. I added a unit test in `JsonLogEventFormatterTests` and updated the README with your suggestion.
If you have additional suggestions feel free to ask further questions.

Otherwise, I'm looking forward for a merge and release 😄

ckadluba commented:
Good work! Thanks a lot for the contribution! 🙂
